Validation/HGCalValidation/plugins/HeterogeneousHGCalRecHitsValidator.h

```diff
@@ -0,0 +1,72 @@
+#ifndef _HGCalMaskResolutionAna_h_
+#define _HGCalMaskResolutionAna_h_
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESTransientHandle.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+#include "SimDataFormats/GeneratorProducts/interface/HepMCProduct.h"
+
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/HepMCCandidate/interface/GenParticle.h"
+#include "DataFormats/HepMCCandidate/interface/GenParticleFwd.h"
+#include "DataFormats/HGCRecHit/interface/HGCRecHit.h"
+#include "DataFormats/HGCRecHit/interface/HGCRecHitCollections.h"
+
+#include "RecoLocalCalo/HGCalRecAlgos/interface/RecHitTools.h"
+
+#include "SimDataFormats/Track/interface/SimTrack.h"
+#include "SimDataFormats/Track/interface/SimTrackContainer.h"
+#include "SimDataFormats/Vertex/interface/SimVertex.h"
+#include "SimDataFormats/Vertex/interface/SimVertexContainer.h"
+
+#include "Validation/HGCalValidation/interface/validHit.h"
+
+#include "TTree.h"
+#include "TH1F.h"
+
+#include <iostream>
+#include <string>
+
+struct ValidRecHits {
+  std::vector<float> energy;
+  std::vector<float> time;
+  std::vector<float> timeError;
+  std::vector<unsigned int> detid;
+  std::vector<unsigned int> flagBits;
+  std::vector<float> son;
+};
+
+class HeterogeneousHGCalRecHitsValidator : public edm::EDAnalyzer {
```

Comment:
Please use some thread-efficient module type. The use of `TFileService` suggests for `one` with https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkOneModuleInterface#edm_one_SharedResources and `usesResource(TFileService::kSharedResource);`.

Reply:
Did you mean "thread-safe", instead of "thread-efficient"? If not, I did not understand how ```edm::one::SharedResources``` makes the modules more thread-efficient.

Reply:
@Dr15Jones As a suggestion, perhaps one could add [here](https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkOneModuleInterface#edm_one_SharedResources) that the ```"FWCore/Framework/interface/one/EDAnalyzer.h"``` header is required.

Reply:
This module should not inherit from `edm::EDAnalyzer` as those base classes are deprecated. As for the include suggestion, it really isn't needed as one must inherit from edm::one::ED*` in order to use the shared resource call.